MBVM-52: Allow to load search indexes from FTP

yvanzo opened the pull request:
# Implement MBVM-52

Building search indexes takes way too much time for small slave server.

If you have enough available bandwidth to download ~30GB of archived search indexes, this patch helps with fetching search index dumps from FTP and loading these into MB Solr.

```bash
sudo docker-compose run --rm musicbrainz /fetch-dump.sh search
sudo docker-compose run --rm search /load-search-indexes.sh
```

More options are available for each scripts, documented with `--help` option.

yvanzo committed:
MBVM-52: Load search indexes from FTP

Reindexing the whole database from scratch can take a very long time.

This patch helps with loading search indexes from FTP instead.

Loading search indexes takes two steps and requires under 80GB:

1. Fetch dump files from FTP, it requires under 30GB.
   Dump files are saved in the volume `searchdump`.

2. Extract dump files in place of Solr data, it requires under 50GB.
   The Docker image tag `metabrainz/mb-solr:3.1.3` comes with `zstd`.

yvanzo committed:
Revert f9e047df: Allow deriving from mb-solr image

yvanzo committed:
Allow setting MB Solr version via build argument

yvanzo committed:
Split fetching dump off from createdb.sh

SuperSandro2000 commented on build/solr/Dockerfile:
```diff
@@ -0,0 +1,4 @@
+ARG MB_SOLR_VERSION=3.1.3
+FROM metabrainz/mb-solr:${MB_SOLR_VERSION}
+
+COPY [ "scripts/*", "/" ]
```
Just a little nit pick: Wouldn't be ``/usr/local/bin/`` be a nicer place?

yvanzo replied:
Changed for scripts in both `search` and `musicbrainz` services with commit https://github.com/metabrainz/musicbrainz-docker/pull/137/commits/fa1a24e7c990ff151c2d13bcf56cbacbb1ce4e77.

SuperSandro2000 reviewed (commented):
.

yvanzo committed:
Install scripts under /usr/local/bin/ instead of /

Running/executing scripts in musicbrainz/search containers was requiring
to prefix with a slash, which is both unnecessary and barely portable.

This patch installs scripts that are expected to be run from the
command-line (in `musicbrainz` and `search`) under `/usr/loca/bin/`
which is included in `PATH` environment variable by default.

Instructions, scripts, and cron task have been updated accordingly.

Upgrade notes:

If you customized replication schedule,
update it to run `replication.sh` without the leading slash.

yvanzo commented:
Rebased to apply the changes from `build/musicbrainz/`  to `build/musicbrainz-dev/` as well.
